pkg/reconciler/namespace/namespace.go

```diff
@@ -116,14 +115,21 @@ func (c *Controller) reconcileNamespace(ctx context.Context, lclusterName string
 	if clusterName == "" {
 		// Namespace is not assigned to a cluster; assign one.
 		// First, list all clusters.
-		cls, err := c.clusterLister.List(labels.Everything())
+		allClusters, err := c.clusterLister.List(labels.Everything())
 		if err != nil {
 			return err
 		}
 
 		// TODO: Filter out un-Ready clusters so a namespace doesn't
 		// get assigned to an un-Ready cluster.
 
+		var cls []*v1alpha1.Cluster
```

Comment:
Why should we require that the `Cluster` is defined in the same logical cluster as the workloads that target it?

Reply:
I was assuming that because a `Cluster` lives in a `Workspace` (aka logical cluster), that you wouldn't want to assign something to a `Cluster` outside of the `Workspace` in question?

Reply:
I almost assumed the opposite - who is it that manages compute? If the use-case for a user being hermetically sealed into their own `Workspace` is that they get transparent multi-cluster and never think about it, I would expect some other external administrator to be the one determining `Cluster`s, `Location` membership, etc.

Reply:
I think it's probably possible for someone/something (controller) to configure compute for a user in their Workspace, but I don't immediately see a path forward for assigning a resource from Workspace A to a Cluster from any random Workspace?

Reply:
ACK

Reply:
@stevekuznetsov iirc we'll be introducing the concept of `ThingLocation` that is local to each workspace and hides the where and how of Cluster resources (i.e. a Cluster won't have to be in a given Workspace for resources in that Workspace to be scheduled to it).